types/src/data.rs

```diff
@@ -521,14 +520,52 @@ pub struct SequencingLeaf<TYPES: NodeType> {
     pub rejected: Vec<<TYPES::BlockType as Block>::Transaction>,
 
     /// the timestamp the leaf was constructed at, in nanoseconds. Only exposed for dashboard stats
-    #[derivative(PartialEq = "ignore")]
     pub timestamp: i128,
 
     /// the proposer id of the leaf
-    #[derivative(PartialEq = "ignore")]
     pub proposer_id: EncodedPublicKey,
 }
 
+impl<TYPES: NodeType> PartialEq for SequencingLeaf<TYPES> {
+    fn eq(&self, other: &Self) -> bool {
+        let delta_left = match &self.deltas {
+            Either::Left(deltas) => deltas.commit(),
+            Either::Right(deltas) => deltas.clone(),
+        };
```

Comment:
FYI I added a function a while back to `LeafType` `get_deltas_commitment` which just does this match